addons/adyen_platforms/models/adyen_account.py

```diff
@@ -154,40 +172,94 @@ def _compute_full_name(self):
             else:
                 adyen_account_id.full_name = "%s %s" % (adyen_account_id.first_name, adyen_account_id.last_name)
 
+    @api.depends('kyc_tier', 'adyen_kyc_ids', 'account_status')
+    def _compute_state(self):
+        for account in self:
+            if account.account_status == 'inactive' and account.kyc_tier == 0:
+                account.state = 'pending'
+            else:
+                if any([k.status == 'awaiting_data' for k in account.adyen_kyc_ids]):
```

Comment:
Having a listcomp in an `any` is a waste as it requires evaluating the entire list before the `any` can (lazily) look for an item flagged `True`.